### ISSUE TITLE
Using a SentenceTransformer like a PyTorch model

### ISSUE THREAD

**Issue by progsi** [UKPLab/sentence-transformers#2384]
Thanks for this library! I am wondering why it seems so unusual using the models provided here in a way like pytorch? My training loop has a few customizations because of which I would not like to train it using the fit function. For example, I would not like to create the full dataset before but rather loop itemwise and sample the triplets on the fly. I would not like to use the fit function for this usecase.

Thanks in advance!

**Comment by tomaarsen** [comment]
Hello!

Like with most models relying on `torch`, you can use your own custom training loops for it. You can directly call the `forward` of the SentenceTransformer with `input_ids` and `attention_mask`, and then the output to use in your loss(es) is likely `sentence_embedding` from the returned dictionary. Call `.backward()` on whatever loss you want, and you'll be training.

With other words, feel free to use SentenceTransformers as a normal PyTorch model.

- Tom Aarsen
